phpunit/class-parsing-test.php

```diff
@@ -51,8 +55,8 @@ function test_parser_output( $html_filename, $parsed_json_filename ) {
 			}
 		}
 
-		$html            = file_get_contents( $html_path );
-		$expected_parsed = json_decode( file_get_contents( $parsed_json_path ), true );
+		$html            = self::strip_r( file_get_contents( $html_path ) );
```

Comment:
Is it not a concern that we needed to strip these characters for the test cases? How is this handled in real-world use cases? Shouldn't the parser be normalizing this for us?

Reply:
In the tests we are comparing fake JSON objects to static HTML that has source formatting (line breaks). By default Git (like any other revision control) converts the line breaks to "native", i.e. `\r\n` on Windows.

In real-world situations, for example  when strings are extracted from a textarea, the modern browsers use `\n` for line breaks. However it is generally a pretty bad practice to try to match "white space" when looking at HTML source.

Looking at all the `\n` and `\u0020` chars in *.json and *.parsed.json in blocks/test/fixtures, we may be in for some trouble later on :)